### PR TITLE
[CodeCompletion] Small improvements to solver-based code completion

### PR DESCRIPTION
**Pull request by ahoppen**
- Exit early in `applyResultBuilderBodyTransform` after reporing solutions to code completion
- Store ignored arguments as `Expr *` instead of `ConstraintLocator`s – this avoids the construction of `ConstraintLocator`s.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: lib/Sema/CSSimplify.cpp [hunk 1]
```diff
@@ -12470,9 +12470,14 @@ bool ConstraintSystem::recordFix(ConstraintFix *fix, unsigned impact) {
...
   }
 
-  if (isArgumentIgnoredForCodeCompletion(fix->getLocator())) {
-    // The argument was ignored. Don't record any fixes for it.
-    return false;
+  if (hasArgumentsIgnoredForCodeCompletion()) {
+    // Avoid simplifying the locator if the constriant system didn't ignore any
+    // arguments.
+    auto argExpr = simplifyLocatorToAnchor(fix->getLocator());
+    if (isArgumentIgnoredForCodeCompletion(argExpr.dyn_cast<Expr *>())) {
```
Comment (xedin) [inline lib/Sema/CSSimplify.cpp:12477]: Nit: use `getAsExpr` helper instead.

**Review by xedin — APPROVED** [review]

File: lib/Sema/CSSimplify.cpp [hunk 2]
```diff
@@ -12470,9 +12470,14 @@ bool ConstraintSystem::recordFix(ConstraintFix *fix, unsigned impact) {
     log << ")\n";
   }
 
-  if (isArgumentIgnoredForCodeCompletion(fix->getLocator())) {
-    // The argument was ignored. Don't record any fixes for it.
-    return false;
+  if (hasArgumentsIgnoredForCodeCompletion()) {
+    // Avoid simplifying the locator if the constriant system didn't ignore any
```
Comment (LucianoPAlmeida) [inline lib/Sema/CSSimplify.cpp:12474]: Typo `constriant` -> `constraint`?

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
For completeness sake: The previous stress tester run (https://ci.swift.org/view/Pull%20Requests/job/swift-PR-macos-with-sourcekit-stress-tester/42/) did not find any new issues.

**Pull request merged**